omod/src/main/java/org/openmrs/module/rwandahivflowsheet/web/model/AdultHivFlowsheetFormData.java

```diff
@@ -236,11 +237,11 @@ private void addARVRegimenTableRow(StringBuilder builder, List<DrugOrder> drugOr
     		
     		Set<String> reasons = new HashSet<String>();
     		for (DrugOrder dor :drugOrders){
-    			if (dor.getDiscontinuedReason() != null 
-    					 && (((dor.getDiscontinuedDate() != null && dor.getDiscontinuedDate().before(maxEndDate)) 
+    			if (Utils.getDiscontinuedReason(dor) != null 
+    					 && (((dor.getEffectiveStopDate() != null && dor.getEffectiveStopDate().before(maxEndDate)) 
     							 ||   (dor.getAutoExpireDate() != null && dor.getAutoExpireDate().before(maxEndDate))))){
```

Comment:
Same comment here as above, you can do away with the autoexpire check since you are using effective stop date

Reply:
[Done](https://github.com/PIH/openmrs-module-rwandahivflowsheet/pull/2/commits/a2a8256b394ac8c5384adc841c152682cd856677)